R/httr2.R

```diff
@@ -47,11 +67,20 @@ on_load(
 )
 
 on_load(
-  chat_perform_async_stream <- coro::async_generator(function(provider, req) {
+  chat_perform_async_stream <- coro::async_generator(function(
+    provider,
+    req,
+    parent_otel_span = NULL
+  ) {
+    setup_active_promise_otel_span(parent_otel_span)
+
     resp <- req_perform_connection(req, blocking = FALSE)
     on.exit(close(resp))
 
     repeat {
+      # Ensure the span is active for each await/yield point
+      setup_active_promise_otel_span(parent_otel_span)
+
```

Comment:
Q: Is this necessary?

The goal is to make sure httr2 has a parent span and I believe that is done within `req_perform_connection()`.

I don't believe we need to reactivate to help with streaming the result back. 

The only important part is request... not the stream.

Reply:
Indeed, even in the async streaming case, these are 'internal' and do not need the promise domain to be active. I've removed these lines in [01f20e6](https://github.com/tidyverse/ellmer/pull/526/commits/01f20e6108aad0e110e8200f89db7f14f250378f).